Homework week3 aline pereira

alineafpereira opened the pull request.

alineafpereira committed:
Create read.txt

alineafpereira committed:
finish homework week3

alineafpereira committed:
finish week3 homework

alineafpereira committed:
Delete gradecalc.js

yash-kapila commented on Week3/homework/exercise5.js:
```diff
@@ -0,0 +1,19 @@
+    return totalPrice; /*how to format the value so it comes just with numbers after coma?*/
+}
+
+let CartForParty = { /* with the objetc here, every key is a product and their values are the price */ 
```
`const` instead of `let` here would work just fine. 🙂 
The name of the variable `CartForParty` should ideally follow camelCase convention i.e `cartForParty` 🙂

alineafpereira replied:
Done.

yash-kapila commented on Week3/homework/exercise4.js:
```diff
@@ -0,0 +1,20 @@
+"use strict"
+/*the grocery list should contain bananas and milk at first and then we should be adding other items*/
+let groceryItem = ["bananas", "milk"]
+let i = 0;
```
Don't think you need this `i` variable, right?

alineafpereira replied:
I didn't, already took it off.

yash-kapila commented on Week3/homework/exercise4.js:
```diff
@@ -0,0 +1,20 @@
+        groceryItem.shift() /*and here we take the first item out since the length is already = 3*/
+
+    }
+    return `You bought ${groceryItem}!`
```
- `const` instead of `let` and `moreItems` instead of `moreitems` variable please.
- I see you are not using semi-colons consistently. It's okay to either use them completely or not use them at all but try to follow a common pattern please 🙂

alineafpereira replied:
Done.

yash-kapila commented on Week3/homework/exercise3.js:
```diff
@@ -0,0 +1,17 @@
+let jobs = ['Web Developer','Chef de Cuisine','Circus Clown','Lawyer','Designer'];
+
+function getElement(elementArray) {
+    let randomthing = Math.floor(Math.random() * elementArray.length);
```
`const` instead of `let` here please

alineafpereira replied:
done.

yash-kapila commented on Week3/homework/exercise3.js:
```diff
@@ -0,0 +1,17 @@
+    return elementArray[randomthing];
+}
+
+console.log(`You will be a ${getElement(jobs)} in ${getElement(locations)}, and married to ${getElement(partnerNames)} with ${getElement(numChildren)} kids.`)
```
👍 for using template literals here

yash-kapila commented on Week3/homework/exercise2.js:
```diff
@@ -0,0 +1,11 @@
+"use strict"
+
+let dogage = []
```
Err... unused variable here.

alineafpereira replied:
done.

yash-kapila commented on Week3/homework/exercise2.js:
```diff
@@ -0,0 +1,11 @@
+
+let dogage = []
+function calculateDogAge(num){
+    let realdogage = num * 7;
```
`const` instead of `let` and `realDogAge` instead of `realdogage` please. Try to read this article again to revisit topic of variable naming conventions - https://github.com/HackYourFuture/fundamentals/blob/master/fundamentals/naming_conventions.md

yash-kapila commented on Week3/homework/exercise1.js:
```diff
@@ -0,0 +1,19 @@
+function getElement(elementArray) {
+    let randomcompliment = Math.floor(Math.random() * elementArray.length);
+    return elementArray[randomcompliment];
+}
```
Why does it not work when inside the array? You can do something like:

```
function giveCompliment(name) {
  const compliments = [...the array as defined currently];
  const randomIndex = Math.floor(Math.random() * compliments.length);

  return `You are so ${compliments[randomIndex]}, ${name}!`;
}
const name = 'Aline';
console.log(giveCompliments(name));
```

yash-kapila commented on Week3/homework/CreditCardValidate.js:
```diff
@@ -0,0 +1,46 @@
+"use strict"
+
+function validCreditCard (num) {
+    let card = num;
```
Ah. I don't think you need to create an extra element `card` here which is nothing but containing the same value as `num` You can use `num` instead.

yash-kapila commented on Week3/homework/CreditCardValidate.js:
```diff
@@ -0,0 +1,46 @@
+    let cardNumber = card.split("");
+    let cardNumberDif = cardNumber.filter((este, i) => cardNumber.indexOf(este) === i);
+    let cardNumberLen = cardNumberDif.length;
+    let soma = 0;
```
I am not sure what are you trying to achieve here. This is firstly because the variable names you are using are not quite descriptive(`cardNumberDif`, `cardNumberLen` and `soma`) about what values those variables would potentially be holding to.

Also, not sure if I understood correctly your logic of the `filter` operation. What is it that you are trying to achieve?

yash-kapila commented on Week3/homework/CreditCardValidate.js:
```diff
@@ -0,0 +1,46 @@
+    let cardNumberLen = cardNumberDif.length;
+    let soma = 0;
+         /* My intention here is to define that the digits can't be all the same*/
+        if (cardNumberLen = 1) {
```
Ah. Be careful here please. Ideally, we should not be doing any assignment operation i.e `cardNumberLen = 1` inside an if condition as it leads to unexpected results. Perhaps you wanted to do `cardNumberLen === 1` instead?

yash-kapila commented on Week3/homework/CreditCardValidate.js:
```diff
@@ -0,0 +1,46 @@
+                if (soma < 16){
+                return `The sum of the digits of the card ${card} is less than 16!`
+                }
+        }
```
👍 for logic but I don't think `soma` is a correct variable name as it doesn't describe what value it contains.

yash-kapila commented on Week3/homework/CreditCardValidate.js:
```diff
@@ -0,0 +1,46 @@
+            return `Your card number have ${num.length} digits and it needs to have 16 digits`
+        }
+        if (num.length > 16){
+            return `Your card number have ${num.length} digits and it needs to have 16 digits`
```
It can simply be like: `num.length !== 16`, isn't it? 🙂

yash-kapila commented on Week3/homework/CreditCardValidate.js:
```diff
@@ -0,0 +1,46 @@
+        }
+        /*Here we assure that the last digit will be even*/
+        for (let x = i=15; num.length === 16; i++){
+            if (num[15] === x % 2 == 1){
```
You don't really need to execute a loop here, right? Since you are only checking for the last digit's divisibility by 2. 🙂

yash-kapila commented on Week3/homework/CreditCardValidate.js:
```diff
@@ -0,0 +1,46 @@
+        else {
+            return `Your credit card ${num} is correct and approved!`
+        }
+     
```
👍 but I don't see the logic of checking whether the card number contains atleast 2 different digits. Could you please check?

alineafpereira replied:
I guess I could finish all the points about this exercise here. Hope it is correct now :)

yash-kapila reviewed (approved):
Hi @alineafpereira - I have reviewed your homework and am approving the same. However, I do see some areas of improvements in the card validator questions which was a bonus question. Could you please have a look? And don't forget to ask for help if you are stuck somewhere for long. Good luck! 🙂

alineafpereira commented:
Hi, @yash-kapila, how are you? Sorry that I took long to see your considerations here. I spent more time than expected to solve the JS2 homework from last week and went nuts with my schedule. Gonna take a look now. Thanks!

alineafpereira committed:
Homework fixed

Needed to upload them all over again because when I tried just to update them from my git I got an repository error.
